src/aioswitcher/device/tools.py

```diff
@@ -165,11 +167,11 @@ def convert_str_to_devicetype(device_type: str) -> DeviceType:
     return DeviceType.MINI
 
 
-def get_token(username: str, password: str) -> DeviceToken:
-    """Make API call to get a Token by username and password."""
+def get_token(username: str) -> DeviceToken:
+    """Make API call to get a Token by username."""
     token = DeviceToken("")
-    request_url = "https://switcher.co.il/GetKey/GetTok/t.php"
-    request_data = {"username": username, "password": password}
+    request_url = "https://switcher.co.il/GetToken/"
+    request_data = {"email": username, "secret": TOKEN_SECRET}
```

Comment:
I don't understand this. Email is the only requirement for fetching the token from Switcher?!

Reply:
@TomerFi Yes.. Email is enough for them. This token part is only local internal communication with the devices. You can't do anything with this token if you are outside the user's network.

Reply:
We can resolve this once we come to terms here: https://github.com/TomerFi/aioswitcher/pull/764#discussion_r1555854445.

Reply:
@TomerFi just a comment
>Proposed Solution 1 - Why even bother

The details there are incorrect, Many users already have the integration active, we cant' break it for them if they have no action to do. HA allow asking users to provide new auth details (called reauth flow) when integration detects that it needs new data from the user. This is mainly used if you change a password to login to your device or cloud based API.
My plan was that when we detect a device that require a token we ask users to add the token, if we don't detect such device we never ask anything from the user. The problem is that only the integration can trigger this, the user can trigger this flow. So what will happen is that if we trigger this, a user supply a token, we don't know how to validate it, we just store it and later the user finds that he can't control his devices. This will immediately goes into an issue "Why is my device not working" it doesn't matter if you document this in font 36, there is no option for the user to understand where is the problem.

.. but lets assume the user realized he typed a wrong token, the only way for him to fix it now is to remove the integration and add it again, this also removes any entity customizations he did (names, icons) and delete all related automations. This would bring a very bad user experience 

3 in my opinion wouldn't be possible or bring anything.

2 & 4 are acceptable by me (with 4.1 as the most preferred solution, since it can only happen with the user who has the details for his account and if you have them it is the same as you do it from their APP (if I have you user and password I can install the APP on my phone and do it.) the benefit here is that you don't store these details, you ask them from the user get the token and only store the token. The token can not be used later to get the user APP details.

4.2 is useless, I really don't understand Switcher, they have now officially provided anyone a backdoor for their Token logic, anyone who has the request token can ask a token for any valid email, so even if they drop this API, I assume there are already many breached tokens. 

>I assume this is the only solution acceptable in Home Assistant, because we hide the bad part in this module

At first I thought we can put the token here, but when you bump the lib the changes are reviewed, I would not want to be the one who approve such change since we all realize it is insecure. 

but considering the history with Switcher I must say I have low hopes, they provided an API (4.1) and changed it while we where working on it, I am also spending too much time for this, I don't blame anyone, but I would not be able to keep this pace for long. I was aiming to give a push this 2 weeks so we finish this, but it doesn't look like it is goin to happen.

Reply:
@thecode 
> The details there are incorrect...

I appreciate the clarification. I suggest that Home Assitant needs some modifications, i.e., not removing the entire integration for modifying tokens. But I assume there's a reason for that, and changing it is more challenging than it sounds. Regardless, I understand and see your point.

> 3 in my opinion wouldn't be possible or bring anything.

It can help verify some typos, at the very least. But I agree with you; after you clarify the requirement, this solution needs more merit.

> 2 & 4 are acceptable by me (with 4.1 as the most preferred solution

Yep, I agree with you. Option 2 should be the preferred option; if it is not possible, option 4.1 is the next. Other options should be off the table.

> 4.2 is useless, I really don't understand Switcher. They have now officially provided anyone a backdoor for their Token logic.

Yeah, I was shocked at this, too. It's like we care about the security and privacy of their devices more than they do. Part of me wants to believe that maybe they didn't think it'd end up as literal in our code base.

---

@YogevBokobza, our next step is to check with Switcher for their preference. Either they provide us with an endpoint for verifying email and tokens (option 2) or an endpoint for fetching tokens per user and password (option 4.1). We're good with both, but prefer the former.

Reply:
> Do we know if the Switcher intends to change this situation? Are there any plans to return an "unauthorized" response for failed tokens?
I believe that Switcher does not intend to change this situation. They return "Invalid credentials" with the current implementation API.

Option 2 I feel like it's much like the option I suggested here in the [comment](https://github.com/TomerFi/aioswitcher/pull/764#discussion_r1559267304)
I think it's durable, I will reach Switcher for that.

Option 4.1 is too risky for them, I don't see it happening.

If option 2 is done by Switcher we can continue? Can we reach this new API in this library?

Reply:
So based on option 2 Switcher made that new API call.
I made the relevant changes converted get_token to validate_token with docs and tests and removed the secrets.
@TomerFi @thecode Please see if that's ok and we can continue :)

Reply:
I'm waiting for @thecode to have a look before resolving this.

Reply:
@TomerFi Hallelujah 😃 
You can resolve and merge :)